### PR TITLE
Add table aws_oam_link closes #1455

### LINKED ISSUE #1455: Add table aws_oam_link
**References**
https://github.com/aws/aws-sdk-go-v2/blob/main/service/oam/api_op_ListLinks.go#L20

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select * from aws_aab.aws_oam_link
+--------------------------------------+------------------------------------------------------------------------------+------------------------------------------------------------------------------+------
| id                                   | arn                                                                          | sink_arn                                                                     | label
+--------------------------------------+------------------------------------------------------------------------------+------------------------------------------------------------------------------+------
| e7ba7e47-c843-4abc-85cd-3992866ce5ad | arn:aws:oam:us-east-1:834793684356:link/e7ba7e47-c843-4abc-85cd-3992866ce5ad | arn:aws:oam:us-east-1:347102155630:sink/c7bf4c81-7bdf-4cbd-a4b5-5dfc61b1489f | aab  
+--------------------------------------+------------------------------------------------------------------------------+------------------------------------------------------------------------------+------

> select * from aws_aab.aws_oam_link where arn = 'arn:aws:oam:us-east-1:834793684356:link/e7ba7e47-c843-4abc-85cd-3992866ce5ad'
+--------------------------------------+------------------------------------------------------------------------------+------------------------------------------------------------------------------+------
| id                                   | arn                                                                          | sink_arn                                                                     | label
+--------------------------------------+------------------------------------------------------------------------------+------------------------------------------------------------------------------+------
| e7ba7e47-c843-4abc-85cd-3992866ce5ad | arn:aws:oam:us-east-1:834793684356:link/e7ba7e47-c843-4abc-85cd-3992866ce5ad | arn:aws:oam:us-east-1:347102155630:sink/c7bf4c81-7bdf-4cbd-a4b5-5dfc61b1489f | aab  
+--------------------------------------+------------------------------------------------------------------------------+------------------------------------------------------------------------------+------

```
</details>

### CODE REVIEW

**Comment by ParthaI** [comment]
Because of the service client must be merged first, Go lang lint is expected to fail for this PR.
Dependency PR - https://github.com/turbot/steampipe-plugin-aws/pull/1495

File: aws/table_aws_oam_link.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+			Hydrate: getAwsOAMLink,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listAwsOAMLinks,
+		},
+		GetMatrixItemFunc: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "id",
+				Description: "The random ID string that Amazon Web Services generated as part of the link ARN.",
```
Comment (misraved) [inline aws/table_aws_oam_link.go:33]:
```suggestion
				Description: "The random ID string that Amazon Web Service generates as part of the link ARN.",
```

File: aws/table_aws_oam_link.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+)
+
+func tableAwsOAMLink(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_oam_link",
+		Description: "AWS OAM Link",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("arn"),
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"ResourceNotFoundException"}),
```
Comment (misraved) [inline aws/table_aws_oam_link.go:22]:
```suggestion
				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"InvalidParameterException", "ResourceNotFoundException"}),
```

Comment (misraved) [reply aws/table_aws_oam_link.go:22]: I think we missed adding `InvalidParameterException` as mentioned in https://docs.aws.amazon.com/OAM/latest/APIReference/API_ListLinks.html

Comment (misraved) [reply aws/table_aws_oam_link.go:22]: Do we need to add a retry config for the `InternalServiceFault` error as mentioned in https://docs.aws.amazon.com/OAM/latest/APIReference/API_ListLinks.html?

File: docs/tables/aws_oam_link.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Table: aws_oam_link
+
+Amazon CloudWatch Observability Access Manager (OAM) link share their observability data with the monitoring account. The shared observability data can include metrics in Amazon CloudWatch, logs in Amazon CloudWatch Logs, and traces in AWS X-Ray.
```
Comment (misraved) [inline docs/tables/aws_oam_link.md:3]:
```suggestion
Amazon CloudWatch Observability Access Manager (OAM) link shares their observability data with the monitoring account. The shared observability data can include metrics in Amazon CloudWatch, logs in Amazon CloudWatch Logs, and traces in AWS X-Ray.
```

File: docs/tables/aws_oam_link.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  l.id,
+  l.arn
+  s.name as sink_name,
+  l.sink_arn
+from
+  aws_oam_link as l,
+  aws_oam_sink as s;
+```
+
+### List links that share data for loggroup resource type
```
Comment (misraved) [inline docs/tables/aws_oam_link.md:33]:
```suggestion
### List links that share data for Cloudwatch log group resource type
```

**Review by misraved — COMMENTED** [review]
Please take a look at the review comments. Thanks!!

File: aws/table_aws_oam_link.go (hunk 2 above)
Comment (ParthaI) [reply aws/table_aws_oam_link.go:22]: I don't think we should add `InternalServiceFault` to ignore config, but we should add `InvalidParameterException` to ignore config

**Review by misraved — APPROVED** [review]

**Pull request merged**